### PR TITLE
KFSPTS-19822 update PMW tax mapping

### PR DESCRIPTION
**Pull request by jhulslander**

### CODE REVIEW

**Review by nkimble — CHANGES_REQUESTED** [review]
For the most part this change looks good but I have found a few things to consider which might help to provide clarity to other developers.

File: src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java [hunk 1]
```diff
@@ -221,29 +221,48 @@ public String getKfsTaxTypeCodeAsString() {
...
     public static final int LLC_TAXED_AS_PARTNERSHIP_TAX_CLASSIFICATION_INDICATOR  = 7;
     public static final int OTHER_TAX_CLASSIFICATION_INDICATOR  = 8;
+    
+    public static final class VendorOwnershipCodes {
+    	public static final String PARTNERSHIP = "PT";
+    	public static final String CORPORATION = "CP";
+    	public static final String S_CORPORATION = "SC";
+    	public static final String ESTATE_TRUST = "ET";
+    	public static final String INDIVIDUAL = "ID";
+    }
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:231]: Sure, I can add type

File: src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java [hunk 2]
```diff
@@ -221,29 +221,48 @@ public String getKfsTaxTypeCodeAsString() {
...
+        C_CORPORATION(C_CORPORATION_TAX_CLASSIFICATION_INDICATOR, "C Corporation", VendorOwnershipCodes.CORPORATION),
+        S_CORPORATION(S_CORPORATION_TAX_CLASSIFICATION_INDICATOR, "S Corporation", VendorOwnershipCodes.S_CORPORATION),
+        PARTNERSHIP(PARTNERSHIP_TAX_CLASSIFICATION_INDICATOR, "Partnership", VendorOwnershipCodes.PARTNERSHIP),
+        TRUST_ESTATE(TRUST_ESTATE_TAX_CLASSIFICATION_INDICATOR, "Trust/estate", VendorOwnershipCodes.ESTATE_TRUST), 
+        LLC_TAXED_AS_C_CORPORATION(LLC_TAXED_AS_C_CORPORATION_TAX_CLASSIFICATION_INDICATOR, "LLC taxed as C Corporation", 
+        		VendorOwnershipCodes.CORPORATION, VendorOwnershipCodes.PARTNERSHIP) ,
+        LLC_TAXED_AS_S_CORPORATION(LLC_TAXED_AS_S_CORPORATION_TAX_CLASSIFICATION_INDICATOR, "LLC taxed as S Corporation", 
+        		VendorOwnershipCodes.S_CORPORATION, VendorOwnershipCodes.PARTNERSHIP),
+        LLC_TAXED_AS_PARTNERSHIP(LLC_TAXED_AS_PARTNERSHIP_TAX_CLASSIFICATION_INDICATOR, "LLC taxed as Partnership", 
+        		VendorOwnershipCodes.PARTNERSHIP, VendorOwnershipCodes.PARTNERSHIP),
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:245]: Per Mary's notes on https://jira.cornell.edu/browse/KFSPTS-19823 .... yes, that is what is desired

File: src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java [hunk 3]
```diff
@@ -221,29 +221,48 @@ public String getKfsTaxTypeCodeAsString() {
...
         OTHER(OTHER_TAX_CLASSIFICATION_INDICATOR, "Other", "OT");
         
         public final int pmwCode;
         public final String pmwDescription;
-        public final String translationToKfsOwnershipTypeCode;
+        public final String legacyTranslationToKfsOwnershipTypeCode;
+        public final String foreignTranslationToKfsOwnershipTypeCode;
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:252]: it is used for the "new" form that supports both form and domestic vendors .... we used this naming convention in other areas in the update of the PMW integration

[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:252]: I am hesitant to get away from legacy and foreign so the naming is similar

File: src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:245]: Thanks for confirming.

File: src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:252]: IMHO, "legacy" would pertain to the old PMW form while "domestic" or "foreign" would pertain to the type of tax classification.

[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:252]: how about using legacyFormTranslationToKfsOwnershipTypeCode and foreignFormTranslationToKfsOwnershipTypeCode for names?  I can understand the confusion, but I am worried about adding more confusion if we get away from the terms legacy and foreign like we used through this update

[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:252]: Those names do provide more clarity while maintaining consistency with this larger update. I am good with them.

File: src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:231]: One more thing that I just saw. "C_" was dropped. Asking, should we maintain that?

[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/PaymentWorksConstants.java:231]: oh, good catch, I will add

**Comment by jhulslander** [comment]
This is ready for review again, I made changes based on the feedback

**Review by nkimble — APPROVED** [review]
My concerns have been addressed. Unit tests have passed. Merging.

**Pull request merged**